microsoft/PowerToys
# [Setup] Centralize registry manipulation and clean up installation logic

## Summary of the Pull Request

**What is this about:**
- introduce high-level registry API which could be used both in modules and the installer. it's simple to use from new modules which require a lot of shellex handlers, also simple to switch from perMachine to perUser installation
- wire up the API with PowerPreview, cleanup and clarify hardcoded CLSIDs
- cleanup PowerPreview logic and remove obsolete unit tests
- since PowerRename and ImageResizer do not modify the registry when toggling them on and off (they read .json setting file directly from a context menu handler callback instead), we don't need to move the registry from .wxs to a custom action, however, it could be easily done if it's preferred
- now bootstrapper always detects if it's not elevated and elevates itself to avoid 2 UAC prompts on update (one for uninstall current version, and another for installing new one)

**What is include in the PR:** 

**How does someone test / validate:** 
- install PT using bootstrapper, without launching PT verify that HKCR\CLSID contains CLSIDs from `modulesRegistry.h`, e.g. `ddee2b8a-6807-48a6-bb20-2338174ff779`
- toggle PowerPreview modules on and off and make sure they work (thumbnails take a while before updating, you try cleaning cache with `rm -Force $env:LOCALAPPDATA\microsoft\windows\explorer\thumbcache*` instead of rebooting)
- also try the steps above with UAC-enabled and as a non-admin user
## Quality Checklist

- [x] **Linked issue:** #10126

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
